src/hooks/journals/useJournalData.ts

```diff
@@ -21,54 +23,66 @@ const useJournalsForCollection = (collectionName: string | undefined) => {
     const [attempts, { inc: incAttempts, reset: resetAttempts }] =
         useCounter(0);
 
-    const { data: gatewayConfig, refresh: refreshAuthToken } =
-        useGatewayAuthToken(collectionName ? [collectionName] : []);
+    const refreshAuthToken = useJournalStore((state) => state.getAuthToken);
+    const brokerAddress = useJournalStore(
+        (state) => state.collectionBrokerAddress
+    );
+    const brokerToken = useJournalStore((state) => state.collectionBrokerToken);
 
     const journalClient = useMemo(() => {
-        if (gatewayConfig?.gateway_url && gatewayConfig.token) {
-            const authToken = gatewayConfig.token;
-            const baseUrl = new URL(gatewayConfig.gateway_url);
+        if (brokerAddress && brokerToken) {
+            const baseUrl = new URL(brokerAddress);
 
-            return new JournalClient(baseUrl, authToken);
+            return new JournalClient(baseUrl, brokerToken);
         } else {
             return null;
         }
-    }, [gatewayConfig]);
+    }, [brokerAddress, brokerToken]);
 
     const fetcher = useCallback(
         async (_url: string) => {
-            if (journalClient && collectionName) {
+            if (
+                journalClient &&
+                collectionName &&
+                brokerAddress &&
+                brokerToken
+            ) {
                 const journalSelector = new JournalSelector().collection(
                     collectionName
                 );
 
-                const dataPlaneListResponse = await dataPlaneFetcher_list(
-                    journalClient,
-                    journalSelector,
-                    'JournalData'
+                const dataPlaneListResponse = await getJournals(
+                    brokerAddress,
+                    brokerToken,
+                    {
+                        include: journalSelector.toLabelSet(),
+                    }
                 );
 
-                if (!Array.isArray(dataPlaneListResponse)) {
+                if (isEmpty(dataPlaneListResponse)) {
                     return Promise.reject(dataPlaneListResponse);
                 }
 
                 return {
                     journals:
-                        dataPlaneListResponse.length > 0
-                            ? dataPlaneListResponse
+                        dataPlaneListResponse.result.journals &&
+                        dataPlaneListResponse.result.journals.length > 0
+                            ? dataPlaneListResponse.result.journals
                             : [],
```

Comment:
If we know that the `dataPlaneListResponse.result.journals` is always an array then I think we could just always return it. Cause even if it is an empty array then that is basically what we're defaulting here.